apps/wallet/src/ui/app/hooks/useRestrictedGuard.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useQuery } from '@tanstack/react-query';
+import { useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+export const RESTRICTED_ERROR = {
+    status: 403,
+    body: 'error code: 1020',
+};
+
+export function useRestrictedGuard() {
+    const navigate = useNavigate();
+    const location = useLocation();
+
+    const { data } = useQuery(
+        ['restricted-guard'],
+        async () => {
+            // NOTE: We use fetch directly here instead of the RPC layer because we don't want this instrumented,
+            // and we also need to work with the response object directly.
+            const res = await fetch('https://wallet-rpc.testnet.sui.io/', {
+                method: 'POST',
+                body: JSON.stringify({
+                    id: 1,
+                    method: 'sui_getLatestCheckpointSequenceNumber',
+                    jsonrpc: '2.0',
+                    params: [],
+                }),
+                headers: {
+                    // Reseting accept makes the response non-HTML
+                    accept: '',
+                    'content-type': 'application/json',
+                },
+            });
+
+            if (res.status === RESTRICTED_ERROR.status) {
+                const body = await res.text();
+                return { restricted: body === RESTRICTED_ERROR.body };
```

Comment:
I wish there was a slightly better way to check this but eh

Reply:
We can change the response body eventually, CloudFlare is actively rolling it out. But I figure this is probably fine for now at least.